yeat/config/config.py

```diff
@@ -48,23 +48,26 @@ def check_valid_keys(self, observed_keys, expected_keys):
             message = f"Found unsupported configuration key(s) '{key_str}'"
             raise AssemblyConfigError(message)
 
-    def create_sample_and_assembly_objects(self):
+    def create_sample_objects(self):
+        samples = {}
         for label, sample in self.config["samples"].items():
-            self.samples[label] = Sample(label, sample)
+            samples[label] = Sample(label, sample)
+        return samples
+
+    def create_assembly_objects(self):
+        assemblies = {}
         for label, assembly in self.config["assemblies"].items():
-            self.assemblies[label] = Assembly(label, assembly, self.threads)
+            assembly["samples"] = self.get_sample_objects(assembly["samples"])
+            assemblies[label] = Assembly(label, assembly, self.threads, self.bandage)
+        return assemblies
+
+    def get_sample_objects(self, samples):
+        return dict(((sample, self.samples[sample]) for sample in samples))
 
-    def validate_samples_to_assembly_modes(self):
-        for assembly_label, assembly in self.assemblies.items():
-            check = False
-            for sample_label in assembly.samples:
-                sample = self.samples[sample_label]
-                if assembly.mode in ["paired", "single"]:
-                    check = True if assembly.mode in sample.sample else False
-                elif assembly.mode == "pacbio":
-                    check = True if set(PACBIO_READS).intersection(sample.sample) else False
-                elif assembly.mode == "oxford":  # pragma: no cover
-                    check = True if set(OXFORD_READS).intersection(sample.sample) else False
-            if check == False:
-                message = f"No samples can interact with assembly mode '{assembly.mode}' for '{assembly_label}'"
-                raise AssemblyConfigError(message)
+    def get_expected_files(self):
+        expected_files = []
+        for sample in self.samples.values():
+            expected_files += sample.expected_files
+        for assembly in self.assemblies.values():
+            expected_files += assembly.expected_files
+        return expected_files
```

Comment:
This function loops through sample and assembly objects and gets all the calculated final files for snakemake input all.

Reply:
Looks good. But maybe could be even more concise?

```python
from itertools import chain

# ...

    def get_expected_files(self):
        expected_files = []
        for element in chain(self.samples.values(), self.assemblies.values()):
            expected_files += element.expected_files
        return expected_files
```

Reply:
This is awesome! I totally forgot about `chain()`!